src/cmd-test-coreos-installer

```diff
@@ -0,0 +1,101 @@
+#!/bin/bash
+# Automate an end-to-end run of coreos-installer with the metal image, which then
+# boots and writes a success message to a virtio-serial port, which we read on the host.
+set -euo pipefail
+
+set -x
+
+dn=$(dirname "$0")
+# shellcheck source=src/cmdlib.sh
+. "${dn}"/cmdlib.sh
+
+build=${1:-latest}
+
+arch=$(arch)
+
+builddir=builds/${build}/${arch}
+buildmeta=${builddir}/meta.json
+
+buildid=$(jq -er .buildid < ${buildmeta})
+metalimg=$(jq -er .images.metal.path < ${buildmeta})
+instkernel=$(jq -er .images.kernel.path < ${buildmeta})
+instinitramfs=$(jq -er .images.initramfs.path < ${buildmeta})
+
+ignition_version=$(disk_ignition_version "${metalimg}")
+
+# OK, we seem to have the images.  Launch our temporary webserver.
+statedir=tmp/coreos-installer-test
+rm -rf "${statedir}" && mkdir -p "${statedir}"
+tftpdir=${statedir}/tftp
+pxeconfigdir=${tftpdir}/pxelinux.cfg
+mkdir -p "${pxeconfigdir}"
+
+# FIXME when we rewrite this in mantle, use an auto-allocated port
+port=8723
+cd "${tftpdir}"
+setpriv --pdeathsig SIGTERM -- kola http-server --port "${port}" &
+cd -
+
+case "${metalimg}" in
+  *.raw) echo "NOTICE: Metal image is not compressed, doing so now temporarily"
```

Comment:
Ahh, the new installer supports uncompressed payloads as well. But yeah, legacy one doesn't. Anyway, worth starting with the more common scenario.